File: trunk/tests/CoreTests/Constructors/Constructors.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SharpKit.JavaScript;

namespace CoreTests.Constructors
{
    [JsType(JsMode.Prototype)]
    class Constructors
    {
        static Constructors()
        {
            var x = 7;
        }

        public Constructors()
        {
            
        }

        void foo()
        {
            var ctor = JsContext.CtorOf<Constructors>();
            var obj = JsContext.JsCode("new ctor()");
        }
    }
}

File: trunk/tests/CoreTests/Constructors/Constructors.js
if (typeof(CoreTests) == "undefined")
    var CoreTests = {};
if (typeof(CoreTests.Constructors) == "undefined")
    CoreTests.Constructors = {};
CoreTests.Constructors.Constructors = function ()
{
};
var x = 7;
CoreTests.Constructors.Constructors.prototype.foo = function ()
{
    var ctor =  CoreTests.Constructors.Constructors;
    var obj =  new ctor();
};
